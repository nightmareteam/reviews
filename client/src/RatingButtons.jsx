import React from 'react';
//import styled from 'styled-components';
const styled = window.styled;
import HelpfulButton from './HelpfulButton.jsx';

/**
 * This component displays the helpfulness buttons (Yes / No / Funny)
 * @param {Object} helpfuleness - an object with counts for each helpfulness property
 * @param {Function} updateHelpfulness - a function to toggle the helpfuless rating
 */
const RatingButtons = ({helpfulness, updateHelpfulness}) => {
  return (
    <Wrapper>
      <HelpfulButton active={helpfulness.yes} updateHelpfulness={updateHelpfulness} string={'Yes'}/>
      <HelpfulButton active={helpfulness.no} updateHelpfulness={updateHelpfulness} string={'No'}/>
      <HelpfulButton active={helpfulness.funny} updateHelpfulness={updateHelpfulness} string={'Funny'}/>
    </Wrapper>
  );
};

const Wrapper = styled.div`
  display: inline
`;

export default RatingButtons;