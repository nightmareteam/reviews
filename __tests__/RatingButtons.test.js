/* eslint-disable camelcase */
import React from 'react';
import { shallow } from 'enzyme';
import RatingButtons from '../client/src/RatingButtons.jsx';
import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-15.4';

Enzyme.configure({ adapter: new Adapter() });

let wrapper;
let props;
let mockCallback;

const createProps = (mockCallback) => ({
  helpfulness: {
    yes: false,
    no: false, 
    funny: false
  },
  updateHelpfulness: mockCallback
});

beforeEach(() => {
  jest.resetModules();
  mockCallback = jest.fn();
  props = createProps(mockCallback);
  wrapper = shallow(<RatingButtons {...props} /> );
});

describe('<RatingButtons /> rendering', () => {
  it('should render elements', () => {
    expect(wrapper).toMatchSnapshot();
  });
});