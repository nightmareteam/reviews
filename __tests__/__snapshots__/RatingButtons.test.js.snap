// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RatingButtons /> rendering should render elements 1`] = `
<styled.div>
  <HelpfulButton
    active={false}
    string="Yes"
    updateHelpfulness={[MockFunction]}
  />
  <HelpfulButton
    active={false}
    string="No"
    updateHelpfulness={[MockFunction]}
  />
  <HelpfulButton
    active={false}
    string="Funny"
    updateHelpfulness={[MockFunction]}
  />
</styled.div>
`;
